docs/source/example_notebooks/dowhy_refuter_assess_overlap.ipynb

```diff
@@ -0,0 +1,508 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "dee3f922",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "refute = model.refute_estimate(\n",
+    "    identified_estimand, \n",
+    "    estimate, \n",
+    "    method_name='assess_overlap', \n",
+    "    support_only=True,\n",
+    "    support_config=support_config,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "0a0c9bda",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Rules cover 100.0% of all samples\n",
+      "\n",
+      "DETAILED RULES:\n",
+      "SUPPORT Rules: Found 2 rule(s), covering 100.0% of samples\n",
+      "\t    Rule #0: (not X1)\n",
+      "\t\t [Covers 75.0% of samples]\n",
+      "\t OR Rule #1: (X1)\n",
+      "\t\t AND (not X2)\n",
+      "\t\t [Covers 25.0% of samples]\n",
+      "No Overlap Rules Fitted (support_only=True).\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(refute)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "9c1e4809",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "refute = model.refute_estimate(\n",
+    "    identified_estimand, \n",
+    "    estimate,\n",
+    "    method_name='assess_overlap', \n",
+    "    overlap_only=True, \n",
+    "    overlap_config=overlap_config\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "51db4ee8",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Rules cover 50.0% of all samples\n",
+      "Overall, 50.0% of samples meet the criteria for inclusion in the overlap set, defined as: Covered by support rules and propensity score in (0.10, 0.90)\n",
+      "Rules capture 100.0% of samples which meet these criteria\n",
+      "\n",
+      "DETAILED RULES:\n",
+      "No Support Rules Fitted (overlap_only=True).\n",
+      "OVERLAP Rules: Found 1 rule(s), covering 50.0% of samples\n",
+      "\t    Rule #0: (not X1)\n",
+      "\t\t AND (not X2)\n",
+      "\t\t [Covers 50.0% of samples]\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(refute)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6ab1255c",
+   "metadata": {},
+   "source": [
+    "### Configuration\n",
+    "\n",
+    "OverRule can be configured by passing in the following arguments\n",
+    "* `overlap_eps`: This should be a number $\\epsilon$ in (0, 0.5) which defines the overlap region as those points where $P(T = 1 \\mid X)$ is in $(\\epsilon, 1 - \\epsilon)$.\n",
+    "\n",
+    "* `cat_feats`: A list of any categorical features (automatically inferred if dtype is object).\n",
+    "\n",
+    "* `support_config`: A `SupportConfig` (see below), which controls optimization parameters\n",
+    "\n",
+    "* `overlap_config`: An `OverlapConfig` (see below), which controls optimization parameters"
+   ]
```

Comment:
I've added a Lalonde example to the notebook.  However, the experimental sample from Lalonde isn't a great test bed by itself, since it's an experiment - in fact, I show in the notebook that overlap holds for all samples :) 

I've included something like what we show in the paper - if you combined the experimental samples from Lalonde with observational controls from another dataset, OverRule does a good job at identifying a resulting lack of overlap.